client/config.go

```diff
@@ -61,3 +61,23 @@ func ValidateChainID(baseCmd *cobra.Command) *cobra.Command {
 	baseCmd.RunE = validateFn
 	return baseCmd
 }
+
+// GenerateChainID wraps a cobra command with a RunE function with base 10 integer chain-id random generation
+// when a chain-id is not provided.
+func GenerateChainID(baseCmd *cobra.Command) *cobra.Command {
+	// Copy base run command to be used after chain verification
+	baseRunE := baseCmd.RunE
+
+	// Function to replace command's RunE function
+	generateFn := func(cmd *cobra.Command, args []string) error {
+		chainID := viper.GetString(flags.FlagChainID)
+
+		if chainID == "" {
+			viper.Set(flags.FlagChainID, ethermint.GenerateRandomChainID())
+		}
```

Comment:
I'd prefer to add this logic to the `ValidateChainID` function

Reply:
But then `ValidateChainID` would not just validate